ravedude/src/main.rs

```diff
@@ -91,44 +100,105 @@ fn main() {
 }
 
 fn ravedude() -> anyhow::Result<()> {
-    let args: Args = structopt::StructOpt::from_args();
+    let mut args: Args = structopt::StructOpt::from_args();
+
+    let manifest_path = 'manifest_path: {
+        // By default Cargo scans the current dir and all of its parents for Cargo.toml,
+        // so we mirror its behavior here.
+        let current_dir = std::env::current_dir()?;
+
+        for dir_to_test in current_dir.ancestors() {
+            let path_to_test = dir_to_test.join(Path::new("Ravedude.toml"));
+            if path_to_test.exists() {
+                break 'manifest_path Some(path_to_test);
+            }
+        }
+
+        None
+    };
+
+    if manifest_path.is_some() {
+        if let Some(board) = args.board.take() {
+            if args.bin.is_none() {
+                // The board arg is taken before the binary, so rearrange the args when Ravedude.toml exists
+                args.bin = Some(std::path::PathBuf::from(board));
+            } else {
+                anyhow::bail!("can't pass board as command-line argument when Ravedude.toml is present; set `board = {:?}` under [general] in Ravedude.toml", board)
+            }
+        }
+    } else if args.board.is_some() && !args.dump_config {
+        warning!(
+            "Passing the board as command-line argument is deprecated, use Ravedude.toml instead:"
+        );
+        eprintln!(
+            "\n# Ravedude.toml\n{}",
+            toml::to_string(&config::RavedudeConfig::from_args(&args)?)?
+        );
+    }
+
+    let mut ravedude_config = match manifest_path.as_deref() {
+        Some(path) => board::get_board_from_manifest(path)?,
+        None => board::get_board_from_name(
+            args
+            .board
+            .as_deref()
+            .ok_or_else(|| anyhow::anyhow!("no board given and couldn't find Ravedude.toml in project, either pass a board as an argument or make a Ravedude.toml."))?
```

Comment:
Actually, don't reference the deprecated board name method in this error message.  We don't want to advertise the legacy mechanism in any way.  Simply state that we couln't find a Ravedude.toml and give a pointer on how to create one.

Reply:
Done :+1:. Currently there isn't a descriptive error message, unfortunately. I'm thinking of something like
```
couldn't find Ravedude.toml in project; see `ravedude --help-toml` for more details
```
or even
```
couldn't find Ravedude.toml in project; see `ravedude doc` for more details
```
The message should lead the user to the documentation as just "create a `Ravedude.toml`" doesn't show the correct format for the manifest. Whatever the documentation happens to be, I'll change the error message to link to it afterwards.